### ISSUE TITLE
Unhandled top-level exception

### ISSUE THREAD

**Issue by gregbostrom** [MinaProtocol/mina#6936]
Version:
```
[DIRTY]4a053a4fba333c61ec4e278f8430f581a439630f
```
coda daemon output:
```
{"timestamp":"2020-12-04 04:10:00.527096Z","level":"Info","source":{"module":"Snark_worker__Functor","location":"File \"src/lib/snark_worker/functor.ml\", line 201, characters 10-21"},"message":"SNARK work $work_ids received from $address. Starting proof generation","metadata":{"address":"127.0.0.1:8301","pid":12853,"process":"Snark Worker","work_ids":[406962543,48690414]}}
2020-12-04 04:10:09 UTC [Fatal] Unhandled top-level exception: $exn
Generating crash report
	exn: {
  "string": "Option.value_exn None",
  "backtrace": [
    "Raised at file \"src/error.ml\" (inlined), line 9, characters 14-30",
    "Called from file \"src/option.ml\", line 69, characters 4-21",
    "Called from file \"src/lib/non_zero_curve_point/non_zero_curve_point.ml\", line 184, characters 6-63",
    "Called from file \"src/lib/non_zero_curve_point/non_zero_curve_point.ml\", line 184, characters 6-63",
    "Called from file \"src/lib/coda_base/signed_command.ml\", line 25, characters 28-40",
    "Called from file \"src/lib/coda_base/signed_command.ml\" (inlined), line 24, characters 6-157",
    "Called from file \"src/lib/coda_base/signed_command.ml\", line 24, characters 6-157",
    "Called from file \"src/lib/coda_base/signed_command.ml\", line 24, characters 6-157",
    "Called from file \"src/lib/coda_base/signed_command.ml\", line 34, characters 4-171",
    "Called from file \"src/lib/coda_base/signed_command.ml\", line 34, characters 4-171",
    "Called from file \"src/lib/coda_base/with_status.ml\", line 6, characters 17-26",
    "Called from file \"src/lib/coda_base/with_status.ml\" (inlined), line 6, characters 4-118",
    "Called from file \"src/lib/coda_base/with_status.ml\", line 6, characters 4-118",
    "Called from file \"src/lib/coda_base/with_status.ml\", line 6, characters 4-118",
    "Called from file \"src/lib/coda_base/transaction_logic.ml\", line 43, characters 14-92",
    "Called from file \"src/lib/coda_base/transaction_logic.ml\", line 42, characters 10-323",
    "Called from file \"src/lib/coda_base/transaction_logic.ml\", line 42, characters 10-323",
    "Called from file \"src/lib/coda_base/transaction_logic.ml\", line 77, characters 18-45",
    "Called from file \"src/lib/coda_base/transaction_logic.ml\", line 77, characters 8-95",
    "Called from file \"src/lib/coda_base/transaction_logic.ml\", line 77, characters 8-95",
    "Called from file \"src/lib/coda_base/transaction_logic.ml\", line 103, characters 8-164",
    "Called from file \"src/lib/coda_base/transaction_logic.ml\", line 103, characters 8-164",
    "Called from file \"src/lib/coda_base/transaction_logic.ml\", line 103, characters 8-164",
    "Called from file \"src/lib/coda_base/transaction_logic.ml\", line 147, characters 8-198",
    "Called from file \"src/lib/coda_base/transaction_logic.ml\", line 147, characters 8-198",
    "Called from file \"src/lib/coda_base/transaction_logic.ml\", line 147, characters 8-198",
    "Called from file \"src/lib/coda_base/transaction_logic.ml\", line 162, characters 49-77",
    "Called from file \"src/lib/coda_base/transaction_logic.ml\", line 161, characters 6-117",
    "Called from file \"src/lib/coda_base/transaction_logic.ml\", line 161, characters 6-117",
    "Called from file \"src/lib/transaction_snark_scan_state/transaction_snark_scan_state.ml\", line 40, characters 10-77",
    "Called from file \"src/lib/transaction_snark_scan_state/transaction_snark_scan_state.ml\", line 39, characters 6-599",
    "Called from file \"src/lib/transaction_snark_scan_state/transaction_snark_scan_state.ml\", line 39, characters 6-599",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 69, characters 12-34",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\" (inlined), line 68, characters 8-164",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 68, characters 8-164",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 68, characters 8-164",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 81, characters 8-89",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\" (inlined), line 81, characters 8-89",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 81, characters 8-89",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 81, characters 8-89",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 92, characters 21-63",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\" (inlined), line 92, characters 6-87",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 92, characters 6-87",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 92, characters 6-87",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 225, characters 46-63",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 225, characters 46-63",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 225, characters 46-63",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 225, characters 46-63",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 225, characters 46-63",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 225, characters 46-63",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 225, characters 46-63",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 220, characters 6-226",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 225, characters 14-66",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 225, characters 14-66",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 225, characters 14-66",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 225, characters 14-66",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 225, characters 14-66",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 225, characters 14-66",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 225, characters 14-66",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\" (inlined), line 220, characters 6-226",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 220, characters 6-226",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 220, characters 6-226",
    "Called from file \"src/read.ml\", line 401, characters 23-47",
    "Called from file \"src/read.ml\" (inlined), line 407, characters 16-64",
    "Called from file \"src/list0.ml\" (inlined), line 6, characters 0-48",
    "Called from file \"src/std_internal.ml\" (inlined), line 105, characters 2-83",
    "Called from file \"src/lib/non_empty_list/non_empty_list.ml\", line 7, characters 16-28",
    "Called from file \"src/lib/non_empty_list/non_empty_list.ml\" (inlined), line 7, characters 4-65",
    "Called from file \"src/lib/non_empty_list/non_empty_list.ml\", line 7, characters 4-65",
    "Called from file \"src/lib/non_empty_list/non_empty_list.ml\", line 7, characters 4-65",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 771, characters 12-184",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\" (inlined), line 770, characters 8-340",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 770, characters 8-340",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 770, characters 8-340",
    "Called from file \"src/utils.ml\", line 182, characters 17-70",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\" (inlined), line 787, characters 6-564",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 787, characters 6-564",
    "Called from file \"src/lib/parallel_scan/parallel_scan.ml\", line 787, characters 6-564",
    "Called from file \"src/lib/transaction_snark_scan_state/transaction_snark_scan_state.ml\", line 142, characters 4-169",
    "Called from file \"src/lib/transaction_snark_scan_state/transaction_snark_scan_state.ml\", line 142, characters 4-169",
    "Called from file \"src/lib/transition_frontier/frontier_base/root_data.ml\", line 10, characters 10-68",
    "Called from file \"src/lib/transition_frontier/frontier_base/root_data.ml\", line 9, characters 6-132",
    "Called from file \"src/lib/transition_frontier/frontier_base/root_data.ml\", line 9, characters 6-132",
    "Called from file \"src/lib/transition_frontier/frontier_base/root_data.ml\", line 111, characters 46-72",
    "Called from file \"src/lib/transition_frontier/frontier_base/root_data.ml\", line 111, characters 6-101",
    "Called from file \"src/lib/transition_frontier/frontier_base/root_data.ml\", line 111, characters 6-101",
    "Called from file \"src/option.ml\", line 183, characters 23-28",
    "Called from file \"src/lib/transition_frontier/persistent_frontier/database.ml\", line 217, characters 8-19",
    "Called from file \"src/lib/transition_frontier/persistent_frontier/database.ml\", line 321, characters 4-63",
    "Called from file \"src/lib/transition_frontier/persistent_frontier/worker.ml\", line 45, characters 16-58",
    "Called from file \"src/lib/transition_frontier/persistent_frontier/worker.ml\", line 52, characters 16-20",
    "Called from file \"src/lib/transition_frontier/persistent_frontier/worker.ml\", line 68, characters 4-38",
    "Called from file \"src/lib/transition_frontier/persistent_frontier/diff_buffer.ml\", line 98, characters 29-74",
    "Called from file \"src/lib/transition_frontier/persistent_frontier/sync.ml\" (inlined), line 21, characters 22-55",
    "Called from file \"src/lib/transition_frontier/persistent_frontier/persistent_frontier.ml\", line 136, characters 8-31",
    "Called from file \"src/lib/transition_frontier/transition_frontier.ml\", line 331, characters 4-99",
    "Called from file \"src/lib/transition_handler/processor.ml\", line 62, characters 9-67",
    "Called from file \"src/lib/transition_handler/processor.ml\", line 213, characters 6-195",
    "Called from file \"src/deferred0.ml\", line 56, characters 64-69",
    "Called from file \"src/job_queue.ml\" (inlined), line 131, characters 2-5",
    "Called from file \"src/job_queue.ml\", line 171, characters 6-47"
  ]
}
2020-12-04 04:10:57 UTC [Info] Daemon child process 12689 terminated after receiving signal "sigkill"
2020-12-04 04:10:57 UTC [Error] Child process of kind "Prover" with pid 12689 has terminated
```

**Comment by mrmr1993** [comment]
This should be fixed by #6921.

**Comment by mrmr1993** [comment]
Closing, #6921 has been merged.
